### PR TITLE
Add Task and Targets

### PR DESCRIPTION
**Pull request by BenVillalobos**
https://github.com/rainersigwald/containers/pull/21 and https://github.com/rainersigwald/containers/pull/18 rebased & combined

### CODE REVIEW

**Comment by BenVillalobos** [comment]
Todo; add an end to end test for all the logic

File: Directory.Packages.props [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+  <PropertyGroup>
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageVersion Include="Microsoft.Build.Utilities.Core" Version="17.2.0"/>
+    <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
+    <PackageVersion Include="MSTest.TestAdapter" Version="2.2.8" />
+    <PackageVersion Include="MSTest.TestFramework" Version="2.2.8" />
+    <PackageVersion Include="coverlet.collector" Version="3.1.2" />
+    <PackageVersion Include="GitHubActionsTestLogger" Version="1.4.1" />
```
Comment (baronfel) [inline Directory.Packages.props:11]: nit: this should stay at 2.0.1

File: System.Containers.Tasks/CreateNewImage.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            for(int i = 0; i < filePaths.Length; i++)
+            {
+                Log.LogMessage("File {0} has relative path of {1}", filePaths[i], relativeFilePaths[i]);
+                filesWithPaths.Add((filePaths[i], relativeFilePaths[i]));
+            }
+
+            Layer newLayer = Layer.FromFiles(filesWithPaths.AsEnumerable());
+
+            image.AddLayer(newLayer);
+            image.SetEntrypoint(Entrypoint, EntrypointArgs?.Split(' ').ToArray());
```
Comment (baronfel) [inline System.Containers.Tasks/CreateNewImage.cs:97]: it'll be important to document that the Entrypoint needs to be a fully-rooted path. I could see users doing something like

```xml
<ContainerEntrypoint>$(AppName)</ContainerEntrypoint>
```
and attempting to package up an executable at what is effectively `/$(AppName)`, which won't exist.

Maybe there's some validation we can do here to save them: either checking that a fully-rooted path was provided, or attempting to 'root' a relative or un-rooted path against the `ContainerWorkingDirectory`.

**Review by baronfel — COMMENTED** [review]
couple quick things as I was poking around :)

**Comment by BenVillalobos** [comment]
Cloud-based note to self:

mkdir "C:\temp\dir\newproject"

dotnet new console
dotnet new nugetconfig
dotnet nuget add source "C:\temp\dir\PackageUploads\" -n local-folder
dotnet build "System.Containers.Tasks"
dotnet nuget push "C:\Path\To\Nupkg" --source local-folder
dotnet add package System.Containers.Tasks
dotnet publish --publishprofile=defaultcontainer

**Pull request closed without merging**